tests/api/test_eval.py

```diff
@@ -25,6 +25,24 @@ def test_post_invalid_schema_400(self):
 
         self.assertEqual(expected, result.json)
 
+    def test_post_invalid_data_400(self):
+        input_body = {"input": 400}
+        args_body = {"input": "", "args": [400]}
+
+        input_result = self.simulate_post(self.PATH, json=input_body)
+        args_result = self.simulate_post(self.PATH, json=args_body)
+
+        self.assertEqual(input_result.status_code, 400)
+        self.assertEqual(args_result.status_code, 400)
+
+        expected = {
+            "title": "Request data failed validation",
+            "description": "400 is not of type 'string'"
+        }
+
+        self.assertEqual(expected, input_result.json)
+        self.assertEqual(expected, args_result.json)
```

Comment:
Use a subtest context manager to isolate the assertions. Furthermore, it could use a for loop over the JSON bodies to reduce code redundancy.